### PR TITLE
feat(backup-s3): tag S3 user-agent for attribution

### PR DESCRIPTION
**Pull request by goanpeca**
### What's being changed:

Sets a `weaviate/<version>` product identifier on the S3 user-agent for the `backup-s3` and `usage-s3` modules, so requests carry a stable application tag. This works with Amazon S3 and any S3-compatible object store configured via `AWS_ENDPOINT` (for example Backblaze B2, Cloudflare R2, or MinIO). The change is minimal and additive: it does not alter credentials, endpoint resolution, request behavior, or existing configuration.

Details:
- `backup-s3/client.go`: calls `client.SetAppInfo(build.AppName, productVersion())` on both the shared minio client and the per-request client built from `X-AWS-*` context headers. A small `productVersion()` helper returns `build.Version`, falling back to `"dev"` for unstamped local builds.
- `usage-s3/storage.go`: adds an aws-sdk-go-v2 user-agent middleware (`AddUserAgentKeyValue(build.AppName, version)`) to the default config, with the same `"dev"` fallback.

Both modules reuse the existing `usecases/build` package for the app name and version, so the tag stays consistent with the rest of Weaviate and with the `weaviate_build_info` metric.

### Review checklist

- [ ] Documentation has been updated, if necessary. Link to changed documentation: not necessary. No public API, config surface, or user-facing behavior changes; this only adds a product tag to the outbound S3 user-agent.
- [ ] Chaos pipeline run or not necessary. Link to pipeline: not necessary.
- [ ] All new code is covered by tests where it is reasonable. The only pure, unit-testable addition is the `productVersion()` fallback helper. The `SetAppInfo` / middleware wiring exercises minio and the AWS SDK against a live endpoint, which the existing module tests do not stand up. Happy to add a small table test for `productVersion()` (empty vs stamped `build.Version`) if you'd like it in this PR.
- [ ] Performance tests have been run or not necessary. Not necessary; setting the user-agent is a one-time client-init call and adds no per-request work.

### CODE REVIEW

**Comment by goanpeca** [comment]
I agree with the CLA

File: modules/usage-s3/storage.go [hunk 1]
```diff
@@ -36,7 +39,15 @@ type S3Storage struct {
 
 // NewS3Storage creates a new S3 storage backend
 func NewS3Storage(ctx context.Context, logger logrus.FieldLogger, metrics *common.Metrics) (*S3Storage, error) {
-	cfg, err := config.LoadDefaultConfig(ctx)
+	version := build.Version
+	if version == "" {
+		version = "dev"
+	}
```
[the remark this answers is not in the document]
Comment (goanpeca) [reply modules/usage-s3/storage.go]: Fixed
